app/src/main/java/com/arctouch/codechallenge/viewmodel/HomeViewModel.kt

```diff
@@ -0,0 +1,35 @@
+package com.arctouch.codechallenge.viewmodel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.ViewModel
+import androidx.paging.PagedList
+import com.arctouch.codechallenge.data.Cache
+import com.arctouch.codechallenge.model.Movie
+import com.arctouch.codechallenge.repository.TmdbRepository
+
+class HomeViewModel(private val tmdbRepository: TmdbRepository): ViewModel() {
+
+    var postsLiveData  :LiveData<PagedList<Movie>>
+
+    init {
+        getGenres()
+        val config = PagedList.Config.Builder()
+            .setPageSize(30)
+            .setEnablePlaceholders(false)
+            .build()
+        postsLiveData  = tmdbRepository.initializedPagedList(config).build()
+    }
+
+    fun getUpcomingMovies():LiveData<PagedList<Movie>> = postsLiveData
```

Comment:
Any special reason for you create this function to access your `LiveData`?

Reply:
@arctouch-matheusfaleiro No, I could use postsLiveData